src/PrestaShopBundle/Entity/Repository/ShopRepository.php

```diff
@@ -41,4 +41,24 @@ public function haveMultipleShops()
     {
         return count($this->findAll()) > 1;
     }
+
+    /**
+     * Get a list of shops for a given search term
+     *
+     * @param string $searchTerm
+     *
+     * @return array
+     */
+    public function findBySearchTerm(string $searchTerm): array
+    {
+        $qb = $this->createQueryBuilder('s')
+            ->select(['s', 'sg'])
+            ->join('s.shopGroup', 'sg')
+            ->where('s.active = true')
+            ->andWhere('s.deleted = false')
+            ->andWhere('s.name LIKE :searchTerm')
```

Comment:
Maybe you should use `LOWER(s.name)` and `'%' . strtolower($searchTerm) . '%'` So that your search is not case sensitive

From what I saw on a stackoverflow message you might need to do `$qb->expr()->like('LOWER(s.name)', '?1'),` for the LOWER to work correctly 🤷‍♂️  (but it was in 2013 😅 )

Reply:
done !